src/Functions/FunctionsConversion.h

```diff
@@ -1835,6 +1835,8 @@ class FunctionConvertFromString : public IFunction
     size_t getNumberOfArguments() const override { return 0; }
 
     bool useDefaultImplementationForConstants() const override { return true; }
+    bool canBeExecutedOnDefaultArguments() const override { return false; }
```

Comment:
How this fix problem ? And how this is related to problem ? Please write comment, and maybe simplify reproducible scenario in stateless tests.

Reply:
In query in testcace it calls `transformHeader` on first query stage with LowCarinality column with empty string in dictionary, so parsing is failing. Reason to have it here same as in `FunctionConvert`. Ok, I'll try to make simpler example without JOIN.

Reply:
Simplified example `SELECT parseDateTimeBestEffort(toLowCardinality(materialize('15-JUL-16')))`, added to test